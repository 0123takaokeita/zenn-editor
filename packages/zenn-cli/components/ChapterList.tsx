import Link from "next/link";
import { Chapter } from "@types";

type Props = { bookSlug: string; chapters: Chapter[] };

export const ChapterList: React.FC<Props> = ({ chapters, bookSlug }) => {
  return (
    <div>
      <ul>
        {chapters.map((chapter) => {
          const realPath = `/books/${bookSlug}/${chapter.position}`;
          return (
            <li key={`ch${chapter.position}`}>
              <Link href="/books/[slug]/[position]" as={realPath}>
                <a href={realPath}>
                  {chapter.title || `${chapter.position}.md`}
                </a>
              </Link>
            </li>
          );
        })}
      </ul>
    </div>
  );
};
