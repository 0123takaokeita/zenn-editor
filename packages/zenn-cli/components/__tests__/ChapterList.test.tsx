import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { Chapter } from '@types';
import { ChapterList } from '@components/ChapterList';

const renderer = createRenderer();
const dummyChapter: Chapter = {
  content: 'test chapter',
  free: false,
  position: undefined,
  slug: 'test-chapter-dummy',
  title: 'dummy chapter',
};
const dummyBookSlug = 'dummy_book_slug_name';

describe('<ChapterList />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(
      <ChapterList chapters={[dummyChapter]} bookSlug={dummyBookSlug} />
    );
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
