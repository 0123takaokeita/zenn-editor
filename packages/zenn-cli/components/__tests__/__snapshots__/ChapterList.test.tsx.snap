// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChapterList /> should render and match the snapshot 1`] = `
<div>
  <ol>
    <li
      className="chapter-list-item"
    >
      <Link
        as="/books/dummy_book_slug_name/test-chapter-dummy.md"
        href="/books/[book_slug]/[chapter_filename]"
        passHref={true}
      >
        <a
          className="chapter-list-link"
        >
          <span
            className="chapter-list-title"
          >
            dummy chapter
          </span>
          <span
            className="chapter-list-filename"
          >
            （
            test-chapter-dummy.md
            ）
          </span>
        </a>
      </Link>
    </li>
  </ol>
</div>
`;
